saleor/graphql/app/dataloaders.py

```diff
@@ -35,3 +38,49 @@ def batch_load(self, keys):
             extensions_map[extension.app_id].append(extension)
             app_extension_loader.prime(extension.id, extension)
         return [extensions_map.get(app_id, []) for app_id in keys]
+
+
+def get_app(raw_auth_token) -> Union[None, App]:
+    if raw_auth_token is None:
+        return None
+    tokens = AppToken.objects.filter(token_last_4=raw_auth_token[-4:]).values_list(
+        "app_id", "auth_token"
+    )
+    app_ids = [
+        app_id
+        for app_id, auth_token in tokens
+        if check_password(raw_auth_token, auth_token)
+    ]
+    return App.objects.filter(id__in=app_ids, is_active=True).first()
+
+
+class AppByTokenLoader(DataLoader):
+    context_key = "app_by_token"
+
+    def batch_load(self, keys):
+        last_4s_map = defaultdict(list)
```

Comment:
I will suggest creating more meaningful variable name, that will also explain to what this map is, like `last_4s_to_raw_token_map`

Reply:
Done.